respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/manageuser/profile/ProfileViewModel.kt

```diff
@@ -0,0 +1,165 @@
+package world.respect.shared.viewmodel.manageuser.profile
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.datetime.LocalDate
+import org.jetbrains.compose.resources.getString
+import world.respect.datalayer.oneroster.rostering.model.OneRosterGenderEnum
+import world.respect.shared.generated.resources.*
+import world.respect.shared.navigation.NavCommand
+import world.respect.shared.navigation.ProfileScreen
+import world.respect.shared.navigation.SignupScreen
+import world.respect.shared.navigation.WaitingForApproval
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.shared.viewmodel.app.appstate.ActionBarButtonUiState
+
+class ProfileViewModel(
+    savedStateHandle: SavedStateHandle,
+) : RespectViewModel(savedStateHandle) {
+
+    private val route: ProfileScreen = savedStateHandle.toRoute()
+
+    private val _uiState = MutableStateFlow(ProfileUiState())
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            _uiState.value = when (route.type) {
+                ProfileType.PARENT, ProfileType.Student -> ProfileUiState(
+                    screenTitle = getString(Res.string.your_profile_title),
+                    actionBarButtonName = getString(Res.string.next),
+                    nameLabel = getString(Res.string.your_name_label),
+                    genderLabel = getString(Res.string.your_gender_label),
+                    dateOfBirthLabel = getString(Res.string.your_dob_label),
+                    note = getString(Res.string.your_note)
+                )
+
+                ProfileType.CHILD -> ProfileUiState(
+                    screenTitle = getString(Res.string.child_profile_title),
+                    actionBarButtonName = getString(Res.string.done),
+                    nameLabel = getString(Res.string.child_name_label),
+                    genderLabel = getString(Res.string.child_gender_label),
+                    dateOfBirthLabel = getString(Res.string.child_dob_label),
+                    note = getString(Res.string.child_note)
+                )
+            }
+
+            _appUiState.update { prev ->
+                prev.copy(
+                    actionBarButtonState = ActionBarButtonUiState(
+                        visible = true,
+                        text = uiState.value.actionBarButtonName,
+                        onClick = { onClickSave() }
+                    ),
+                    title = uiState.value.screenTitle,
+                    hideBottomNavigation = true,
+                    userAccountIconVisible = false
+                )
+            }
+        }
+
+    }
+
+
+    fun onFullNameChanged(value: String) {
```

Comment:
On entity property changed must not by asynchronous. Changing a textfield value must be done synchronously. By changing the property types - you won't need the async launch here.